### PR TITLE
Use file compression and the correct separator

### PR DESCRIPTION
**Pull request by Eh2406**
Pandas can write, and Cardo can upload, CSVs faster if the files are compressed.
Also, the os.path.join is a more reliable way to join paths.

### CODE REVIEW

File: cartoframes/context.py [hunk 1]
```diff
@@ -460,15 +460,15 @@ def remove_tempfile(filepath):
             """removes temporary file"""
             os.remove(filepath)
 
-        tempfile = '{temp_dir}/{table_name}.csv'.format(temp_dir=temp_dir,
-                                                        table_name=table_name)
+        tempfile = os.path.join('temp_dir, table_name + '.csv.gz')
```
Comment (andy-esch) [inline cartoframes/context.py]: Great catch, thanks!

Comment (andy-esch) [reply cartoframes/context.py]: Actually, I just noticed there's an extra `'` right before `temp_dir`.

File: cartoframes/context.py [hunk 2]
```diff
@@ -460,15 +460,15 @@ def remove_tempfile(filepath):
...
         self._debug_print(tempfile=tempfile)
         df.drop(labels=[geom_col], axis=1, errors='ignore').to_csv(
                 path_or_buf=tempfile,
                 na_rep='',
                 header=pgcolnames,
                 index=False,
-                encoding='utf-8')
+                encoding='utf-8',
+                compression='gzip')
```
Comment (andy-esch) [inline cartoframes/context.py]: I love this idea, but we've found mixed results in terms of performance. Do you have any benchmarks that compares the non-compressed to the compressed?

I think what we'll do here is add an optional `compression` kwarg, and then do something like follows:

```python
tempfile = os.path.join(temp_dir, '{table_name}.{ext}'.format(table_name=table_name, ext='csv.gz' if kwargs.get('compression') else 'csv')
...
df.drop(...

        compression=kwargs.get('compression')) 
```

File: cartoframes/context.py (hunk 1 above)
Comment (Eh2406) [reply cartoframes/context.py]: good catch, will fix.

File: cartoframes/context.py (hunk 2 above)
Comment (Eh2406) [reply cartoframes/context.py]: I don't have a micro benchmark, just a sense from using the ui. I tend to be uploading large files so perhaps other use cases will have different results. An optional arge would work.

**Comment by Eh2406** [comment]
When this meets your approval, someone should add documentation.

**Comment by andy-esch** [comment]
Looks great, thanks again :)

For now, acceptance of external PRs follows: first a merge to `develop`, where we run the CI tests (which requires authentication against a carto account). Then we do a PR from `develop` to `master` after the tests pass. I'll ping you on that PR, and add to the documentation there too.

**Pull request merged**